akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/ClusterShardingRolePartitioningSpec.scala

```diff
@@ -0,0 +1,171 @@
+      .withFallback(SharedLeveldbJournal.configToEnableJavaSerializationForTest)
+      .withFallback(MultiNodeClusterSpec.clusterConfig))
+
+}
+
+object ClusterShardingMinMembersPerRoleNotConfiguredConfig extends ClusterShardingMinMembersPerRoleConfig {
+
+  nodeConfig(first, second, third)(r1Config)
+
+  nodeConfig(fourth, fifth)(r2Config)
+}
+
+object ClusterShardingMinMembersPerRoleConfiguredConfig extends ClusterShardingMinMembersPerRoleConfig {
+
+  nodeConfig(first, second, third)(
+    ConfigFactory.parseString("akka.cluster.role.R1.min-nr-of-members = 3").withFallback(r1Config))
+
+  nodeConfig(fourth, fifth)(
+    ConfigFactory.parseString("akka.cluster.role.R2.min-nr-of-members = 2").withFallback(r2Config))
+}
+
+abstract class ClusterShardingMinMembersPerRoleNotConfiguredSpec
+    extends ClusterShardingRolePartitioningSpec(ClusterShardingMinMembersPerRoleNotConfiguredConfig)
+
+class ClusterShardingMinMembersPerRoleNotConfiguredMultiJvmNode1
+    extends ClusterShardingMinMembersPerRoleNotConfiguredSpec
+class ClusterShardingMinMembersPerRoleNotConfiguredMultiJvmNode2
+    extends ClusterShardingMinMembersPerRoleNotConfiguredSpec
+class ClusterShardingMinMembersPerRoleNotConfiguredMultiJvmNode3
+    extends ClusterShardingMinMembersPerRoleNotConfiguredSpec
+class ClusterShardingMinMembersPerRoleNotConfiguredMultiJvmNode4
+    extends ClusterShardingMinMembersPerRoleNotConfiguredSpec
+class ClusterShardingMinMembersPerRoleNotConfiguredMultiJvmNode5
+    extends ClusterShardingMinMembersPerRoleNotConfiguredSpec
+
+abstract class ClusterShardingMinMembersPerRoleSpec
+    extends ClusterShardingRolePartitioningSpec(ClusterShardingMinMembersPerRoleConfiguredConfig)
+
+class ClusterShardingMinMembersPerRoleSpecMultiJvmNode1 extends ClusterShardingMinMembersPerRoleSpec
+class ClusterShardingMinMembersPerRoleSpecMultiJvmNode2 extends ClusterShardingMinMembersPerRoleSpec
+class ClusterShardingMinMembersPerRoleSpecMultiJvmNode3 extends ClusterShardingMinMembersPerRoleSpec
+class ClusterShardingMinMembersPerRoleSpecMultiJvmNode4 extends ClusterShardingMinMembersPerRoleSpec
+class ClusterShardingMinMembersPerRoleSpecMultiJvmNode5 extends ClusterShardingMinMembersPerRoleSpec
+
+abstract class ClusterShardingRolePartitioningSpec(config: ClusterShardingMinMembersPerRoleConfig)
+    extends MultiNodeSpec(config)
+    with MultiNodeClusterSpec
+    with ImplicitSender {
+
+  import config._
+
+  override def initialParticipants: Int = roles.size
+
+  private val fourthAddress = node(fourth).address
+  private val fifthAddress = node(fifth).address
+
+  "Cluster Sharding with roles" must {
+
+    "start the cluster, await convergence, init sharding on every node: 2 data types, 'akka.cluster.min-nr-of-members=2', partition shard location by 2 roles" in {
+      awaitClusterUp(first, second, third, fourth, fifth)
+      enterBarrier(s"${roles.size}-roles-up")
+
+      runOn(first, second, third, fourth, fifth) {
+        ClusterSharding(system).start(
```

Comment:
maybe those `start` should be placed before the `awaitClusterUp`?
doesn't matter, but maybe more realistic to use `start` early